keyboards/nuphy/air60_v2/ansi/mcu_pwr.c

```diff
@@ -0,0 +1,443 @@
+    SYSCFG->EXTICR[EXTI_PinSourcex >> 0x02] |= (((uint32_t)EXTI_PortSourceGPIOx) << (0x04 * (EXTI_PinSourcex & (uint8_t)0x03)));
+}
+
+// #include "hal_lld.h"
+#define EXTI_PortSourceGPIOA ((uint8_t)0x00)
+#define EXTI_PortSourceGPIOB ((uint8_t)0x01)
+#define EXTI_PortSourceGPIOC ((uint8_t)0x02)
+#define EXTI_PortSourceGPIOD ((uint8_t)0x03)
+
+#endif
+
+/**
+ * @brief  Light sleep by powering off LEDs.
+ * @note This is Nuphy's "open sourced" sleep logic. It's not deep sleep.
+ */
+void enter_light_sleep(void) {
+    uart_send_cmd(CMD_SET_CONFIG, 5, 5);
+    if ((dev_info.link_mode == LINK_RF_24 && f_rf_sleep) || dev_info.rf_state != RF_CONNECT) {
+        uart_send_cmd(CMD_SLEEP, 5, 5);
+    }
+
+    led_pwr_sleep_handle();
+    // break_all_key();
+    clear_report_buffer_and_queue();
+}
+
+/**
+ * @brief  Enter deep sleep
+ * @note This is Nuphy's un-released logic with some cleanup/refactoring
+ *       The MCU is put on a low power mode.
+ */
+void enter_deep_sleep(void) {
+    //------------------------ RF to sleep
+    enter_light_sleep();
+
+#if (MCU_SLEEP_ENABLE)
+    //------------------------ Turn off USB if not used
+    if (dev_info.link_mode != LINK_USB) {
+        f_usb_deinit = 1;
+        m_deinit_usb_072();
+    }
+
+    // Close timer
+    // if (tim6_enabled) TIM_Cmd(TIM6, DISABLE);
+
+    //------------------------ Configure WakeUp Key
+
+    for (uint8_t i = 0; i < MATRIX_COLS; ++i) {
+        gpio_set_pin_output(col_pins[i]);
+        gpio_write_pin_high(col_pins[i]);
+    }
+    for (uint8_t i = 0; i < MATRIX_ROWS; ++i) {
+        gpio_set_pin_input_low(row_pins[i]);
+    }
+
+    // Configure interrupt source - all 5 rows of the keyboard.
+    SYSCFG_EXTILineConfig(EXTI_PORT_R0, EXTI_PIN_R0);
+    SYSCFG_EXTILineConfig(EXTI_PORT_R1, EXTI_PIN_R1);
+    SYSCFG_EXTILineConfig(EXTI_PORT_R2, EXTI_PIN_R2);
+    SYSCFG_EXTILineConfig(EXTI_PORT_R3, EXTI_PIN_R3);
+    SYSCFG_EXTILineConfig(EXTI_PORT_R4, EXTI_PIN_R4);
+    SYSCFG_EXTILineConfig(EXTI_PORT_R5, EXTI_PIN_R5);
+
+    EXTI_InitTypeDef m_exti;
+    EXTI_StructInit(&m_exti);
+    m_exti.EXTI_Line    = 0XFFFF; // GPIO 0-15
+    m_exti.EXTI_LineCmd = ENABLE;
+    m_exti.EXTI_Mode    = EXTI_Mode_Interrupt;
+    m_exti.EXTI_Trigger = EXTI_Trigger_Falling;
+    EXTI_Init(&m_exti);
+
+    NVIC_InitTypeDef NVIC_InitStructure;
+    NVIC_InitStructure.NVIC_IRQChannel         = EXTI4_15_IRQn;
+    NVIC_InitStructure.NVIC_IRQChannelPriority = 0;
+    NVIC_InitStructure.NVIC_IRQChannelCmd      = ENABLE;
+    NVIC_Init(&NVIC_InitStructure);
+    NVIC_InitStructure.NVIC_IRQChannel = EXTI0_1_IRQn;
+    NVIC_Init(&NVIC_InitStructure);
+    NVIC_InitStructure.NVIC_IRQChannel = EXTI2_3_IRQn;
+    NVIC_Init(&NVIC_InitStructure);
+
+
+    // led_pwr_sleep_handle();
+
+    gpio_set_pin_output(DEV_MODE_PIN);
+    gpio_write_pin_low(DEV_MODE_PIN);
+
+    gpio_set_pin_output(SYS_MODE_PIN);
+    gpio_write_pin_low(SYS_MODE_PIN);
+
+    gpio_set_pin_output(A11);
```

Comment:
The A11 and A12 are connected to the is31fl3733 (for the leds) at least for air96. If it is the same for air60 than sleep should work fine. If it is not connected the same for air60 the keyboard will fail to enter sleep mode.